spring-integration-core/src/main/java/org/springframework/integration/support/utils/IntegrationUtils.java

```diff
@@ -184,9 +184,11 @@ public static RuntimeException wrapInHandlingExceptionIfNecessary(Message<?> mes
 		RuntimeException runtimeException = (ex instanceof RuntimeException)
 				? (RuntimeException) ex
 				: new MessageHandlingException(message, text.get(), ex);
+
 		if (!(ex instanceof MessagingException) ||
 				((MessagingException) ex).getFailedMessage() == null) {
-			runtimeException = new MessageHandlingException(message, text.get(), ex);
+			runtimeException = new MessageHandlingException(message, text.get(),
+					(ex instanceof IllegalStateException && ex.getCause() != null) ? ex.getCause() : ex);
```

Comment:
I think this might be a bit of abuse and we might need to introduce our own exception like `MessagingMethodInvokerException` to be able to let end-user `IllegalStateException` to bubble as is.

Of course if you agree with the change to the `MessagingMethodInvokerHelper`...